ts/session/apis/snode_api/onions.ts

```diff
@@ -310,12 +311,11 @@ export async function processOnionRequestErrorAtDestination({
   destinationSnodeEd25519,
   associatedWith,
   allow401s,
-}: {
+}: WithAllow401s & {
   statusCode: number;
   body: string;
   destinationSnodeEd25519?: string;
   associatedWith?: string;
-  allow401s: boolean;
 }) {
```

Comment:
These keys seem to be used in a few places in different ways but with the same properties. See inside of `ts/session/apis/snode_api/onions.ts`,  I wonder if we should create a `WithX` types for them? Either per property or a combined type since they always seem to be together. Let me know what you think?

```tsx
{
  associatedWith?: string,
  destinationSnodeEd25519?: string
}
```

Reply:
Not strictly related but in `processOnionResponse` I just noticed we have an `any` type for the status code. It might be good type it correctly? See line 602

```tsx
const status = jsonRes.status_code || jsonRes.status;
```

Reply:
I've made a few use the withs, but not all of them can be (like when they are not part of an object, but just args of a function, as a tuple). The others will need to be fixed as part of a bigger refactor.

Reply:
about the any, I added a `    // TODO: type those status
`